pulsar-common/src/main/java/org/apache/pulsar/common/protocol/schema/SchemaStorage.java

```diff
@@ -28,6 +30,20 @@ public interface SchemaStorage {
 
     CompletableFuture<SchemaVersion> put(String key, byte[] value, byte[] hash);
 
+    /**
+     * Put the schema to the schema storage.
+     *
+     * @param key The schema ID
+     * @param fn The function to calculate the value and hash that need to put to the schema storage
+     *           The input of the function is all the existing schemas that used to do the schemas compatibility check
+     * @return The schema version of the stored schema
+     */
+    default CompletableFuture<SchemaVersion> put(String key,
```

Comment:
LGTM! But I have a question why do we need to add a default method when the interface only has one implementation class?

Reply:
@liangyepianzhou 

Users can have their own schema registry implementation 

Please see the broker.conf

```
### --- Schema storage --- ###
# The schema storage implementation used by this broker
schemaRegistryStorageClassName=org.apache.pulsar.broker.service.schema.BookkeeperSchemaStorageFactory
```